consoles/virtio_terminal.pm

```diff
@@ -86,21 +85,20 @@ sub disable {
 
 sub save_snapshot {
     my ($self, $name) = @_;
+    $self->SUPER::save_snapshot($name);
 
-    if (defined($self->{screen})) {
-        $self->{snapshots}->{$name} = $self->{screen}->peak();
-    } else {
-        $self->{snapshots}->{$name} = '';
-    }
+    $self->snapshot($name, 'buffer', $self->{screen} ? $self->{screen}->peak() : $self->{preload_buffer});
 }
 
 sub load_snapshot {
     my ($self, $name) = @_;
+    $self->SUPER::load_snapshot($name);
 
+    my $buffer = $self->snapshot($name, 'buffer') // '';
     if (defined($self->{screen})) {
-        $self->{screen}->{carry_buffer} = $self->{snapshots}->{$name};
+        $self->{screen}->{carry_buffer} = $buffer;
     } else {
-        $self->{preload_buffer} = $self->{snapshots}->{$name};
+        $self->{preload_buffer} = $buffer;
```

Comment:
also see this report about missing coverage of changed lines

Reply:
I get `unauthorized` if I click on https://app.codecov.io/gh/os-autoinst/os-autoinst/compare/1675 (the coverage picture), do I do something wrong? Or could I just get read access?

Reply:
For me it works but one has to login on codecov (via GitHub) first.

Reply:
hm ok, now it is working for me as well. Maybe I wasn't logged in...